### PR TITLE
Dynamic MKL windows

### PR DESCRIPTION
**Pull request by mantaionut**
Updated MKL on Windows to 2021.4.0 in order to have dynamic version on Windows.
Changed the way conda smoke test is done on Windows by creating a local channel in order to test properly that the updated dependency is downloaded.

### CODE REVIEW

File: conda/pytorch-nightly/meta.yaml [hunk 1]
```diff
@@ -37,8 +38,9 @@ requirements:
 
   run:
     - python
+    - mkl==2021.4.0 # [win]
```
Comment (malfet) [inline conda/pytorch-nightly/meta.yaml:41]: This needs to be moved under if not defined cross compiling, isn't it?

File: conda/pytorch-nightly/meta.yaml [hunk 2]
```diff
@@ -19,6 +19,7 @@ requirements:
     - python
     - setuptools
     - pyyaml
+    - mkl-devel=2021.4.0 # [win]
```
Comment (malfet) [inline conda/pytorch-nightly/meta.yaml:22]: Condition should be x86_64 and win

File: conda/pytorch-nightly/meta.yaml [hunk 3]
```diff
@@ -37,8 +38,9 @@ requirements:
 
   run:
     - python
+    - mkl==2021.4.0 # [win]
     {% if cross_compile_arm64 == 0 %}
-    - mkl >=2018 # [x86_64]
+    - mkl==2021.4 # [x86_64]
```
Comment (malfet) [inline conda/pytorch-nightly/meta.yaml:43]: No, one could not make this change, please limit it to not win

**Review by malfet — CHANGES_REQUESTED** [review]

**Comment by mantaionut** [comment]
@malfet I have made the requested changes. Can you please recheck the changes?

**Comment by kit1980** [comment]
@mantaionut please resolve merge conflicts

**Review by malfet — COMMENTED** [review]
Looks ok, but can we have a test plan of sorts (for example by creating PR against PyTorch and running binary builds?)

**Review by malfet — APPROVED** [review]
Thank you for making the change, please resolve merge conflicts, otherwise LGTM

**Comment by mantaionut** [comment]
@malfet I rebased and fix the conflicts. However due to the current python 3.12 conda build which is using mkl=2023.1 I was not able to compile it with mkl 2021.4.0 due to conda conflicts. So I made the change to use mkl 2023.1 on conda py12 while on all the other specifications it remains with mkl 2021.4.0.

**Pull request merged**